### ISSUE TITLE
CMake compilation error with MPI_C and SUNDIALS

### ISSUE THREAD

**Issue by mikekryjak** [boutproject/BOUT-dev#2945]
The recent SUNDIALS update made it necessary to add the C language to the `project` line in the Hermes-3 cmake script (see https://github.com/bendudson/hermes-3/pull/233/commits/7407e35e146425f86756a2e7b5dd6b171832ff47). I'm leaving this here in case this information is useful to someone.

**Comment by Steven-Roberts** [comment]
Just seeing this issue now. Is this still the case after [this PR](https://github.com/boutproject/BOUT-dev/pull/2922) was merged? In particular, with this change https://github.com/boutproject/BOUT-dev/pull/2922/files#diff-3c300e85313a4acb4aa412b9d561a30fd081396a05de144d624ceec30e387c5dR300

**Comment by mikekryjak** [comment]
Yes, we are using this commit: https://github.com/boutproject/BOUT-dev/commit/d7f88073f3ad707d624f8b9fed0a97a49596c542

You can see that the changes are there:
https://github.com/boutproject/BOUT-dev/blob/d7f88073f3ad707d624f8b9fed0a97a49596c542/cmake/SetupBOUTThirdParty.cmake#L300-L301

**Comment by tbody-cfs** [comment]
Possibly related.

Trying to compile with CMake, I'm getting a new error `Reason given by package: MPI component 'C' was requested, but language C is not enabled.` when going from [7d261d4](https://github.com/boutproject/BOUT-dev/tree/7d261d41e60ef5595c93ee7dd9030f28390ad128) to [7bbf7f3](https://github.com/boutproject/BOUT-dev/tree/7bbf7f39b5f3a4c054216015749bf8981a7cf2be)

```
cmake -B ${BOUTPP_BUILD_DIR} -S ${BOUTPP_SRC_DIR} \
-Wno-dev \
-DCMAKE_BUILD_TYPE=RelWithDebInfo \
-DCMAKE_CXX_COMPILER=mpic++ \
-DBOUT_UPDATE_GIT_SUBMODULE=OFF \
-DBOUT_ENABLE_PYTHON=ON \
-DBOUT_USE_SYSTEM_MPARK_VARIANT=OFF \
-DBOUT_USE_HYPRE=OFF \
-DBOUT_USE_PETSC=ON \
-DBOUT_USE_SLEPC=ON \
-DBOUT_USE_SUNDIALS=ON \
-DBOUT_DOWNLOAD_SUNDIALS=ON \
-DBOUT_USE_NLS=OFF \
-DBOUT_ENABLE_METRIC_3D=OFF \
-DBOUT_GENERATE_FIELDOPS=OFF
```

Comparing those commits
https://github.com/boutproject/BOUT-dev/compare/7bbf7f3...7d261d4

**Comment by bendudson** [comment]
Hi @tbody-cfs ! I don't see anything obvious in those commits. Perhaps adding "C" to this line might help?

https://github.com/boutproject/BOUT-dev/blob/next/CMakeLists.txt#L54

so that the project is declared as:
```
project(BOUT++
  DESCRIPTION "Fluid PDE solver framework"
  VERSION ${BOUT_CMAKE_ACCEPTABLE_VERSION}
  LANGUAGES CXX C)
```
?

**Comment by Steven-Roberts** [comment]
I'm not able to replicate this issue, but I think that when SUNDIALS is downloaded and added with `FetchContent_Declare` that doesn't add C to the enabled languages. If @bendudson's suggestion fixes the issue, then that's probably the case. The fix would be moving
https://github.com/boutproject/BOUT-dev/blob/b70f79f3a7d104f3e08b63c3539e4f0d7de1b575/cmake/SetupBOUTThirdParty.cmake#L296
up a level.

**Comment by tbody-cfs** [comment]
If I apply a patch doing what @bendudson is suggesting, I can compile with no issues. I'm not sure if this breaks anything, but I can open a PR to discuss there
